src/Config/Auth.php

```diff
@@ -374,8 +374,23 @@ public function registerRedirect(): string
 
     protected function getUrl(string $url): string
     {
-        return strpos($url, 'http') === 0
-            ? $url
-            : rtrim(site_url($url), '/ ');
+        // To accommodate all url patterns
+        $final_url = '';
+
+        switch ($url) {
+            case strpos($url, 'http') === 0: // URL begins with 'http'. E.g. http://example.com
```

Comment:
This condition is too loose.
If there is a named route starting with `http`, it matches this.

Reply:
If a named route starts with `http`, that means it is an absolute path, so it should be the `$final_url`.

Reply:
I don't think anyone is likely to name like this, but it is possible.:
```
+--------+-------+-----------+------------------------------+----------------+---------------+
| Method | Route | Name      | Handler                      | Before Filters | After Filters |
+--------+-------+-----------+------------------------------+----------------+---------------+
| GET    | /     | http-home | \App\Controllers\Home::index |                | toolbar       |
+--------+-------+-----------+------------------------------+----------------+---------------+
```
If it starts with `http://` or `https://` it is an absolute URL.

Reply:
Ah yes, you are correct.